website/docs/plugin/sdkv2/resources/write-only-attributes.mdx

```diff
@@ -0,0 +1,177 @@
+
+```go
+// List traversal
+cty.GetAttrPath("list_attribute").IndexInt(0)
+
+
+// List nested block traversal
+cty.GetAttrPath("list_nested_block").IndexInt(1).getAttr("list_nested_block_attribute")
+```
+
+Sets or set nested blocks can be traversed using `Index()`. `Index()` takes in a `cty.Value` of the set element that you want to traverse into.
+
+```go
+// Set traversal
+cty.GetAttrPath("set_attribute").Index(cty.String("set_string_val"))
+
+
+// Set nested block traversal
+cty.GetAttrPath("set_nested_block").
+Index(cty.ObjectVal(map[string]cty.Value{"set_nested_block_attribute": cty.StringVal("valueA")}))
+.GetAttr("set_nested_block_attribute") // returns path to cty.String("valueA")
+
+```
+
+### cty.Value
+
+When working with `cty.Value`, you must always check the type of the value before converting it to a Go value or else the conversion will cause a panic.
+
+```go
+// Check that the type is a cty.String before conversion
+if !woVal.Type().Equals(cty.String) {
+    return errors.New("error retrieving write-only attribute: password_wo - retrieved config value is not a string")
+}
+
+// Check if the value is not null
+if !woVal.IsNull() {
+    // Now we can safely convert to a Go string
+    encryptedValue = woVal.AsString()
+}
+```
+
+## PreferWriteOnly Validator
+
+`PreferWriteOnlyAttribute()` is a validator that takes a `cty.Path` to a regular attribute and a `cty.Path` to a write-only attribute.
+
+Use this validator when you have a write-only version of an existing attribute, and you want to encourage practitioners to use the write-only version whenever possible.
+
+The validator returns a warning if the Terraform client is 1.11 or above and the value to the regular attribute is non-null.
+
+Usage:
+
+```go
+func resourceDbInstance() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceCreate,
+		Read:   resourceRead,
+		Delete: resourceDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceImport,
+		},
+		Schema: //omitted for brevity
+		ValidateRawResourceConfigFuncs: []schema.ValidateRawResourceConfigFunc{
+            validation.PreferWriteOnlyAttribute(cty.GetAttrPath("password"), cty.GetAttrPath("password_wo")),
+		},
+	}
+}
+```
+
+```hcl
+resource "example_db_instance" "ex" {
+  username = "foo"
+  password = "bar" # returns a warning encouraging practitioners to use `password_wo` instead.
+}
+```
+
+When using `cty.Path` to traverse into a nested block, use an unknown value to indicate any key value:
+
+- For lists: `cty.Index(cty.UnknownVal(cty.Number))`,
+- For maps: `cty.Index(cty.UnknownVal(cty.String))`,
+- For sets: `cty.Index(cty.UnknownVal(cty.Object(nil)))`,
+
+
+## Best Practices
+
+Since write-only attributes have no prior values, we cannot determine user intent with a write-only attribute alone. To determine when to use/not use a write-only attribute value in your provider, we recommend using other attributes in the provider. For example:
```

Comment:
Not sure the wording 🤔 

```suggestion
Since write-only attributes have no prior values, user intent cannot be determined with a write-only attribute alone. To determine when to use/not use a write-only attribute value in your provider, we recommend using other non-write-only attributes in the provider. For example:
```